src/Features/CSharp/Portable/SimplifyTypeNames/SimplifyTypeNamesCodeFixProvider.cs

```diff
@@ -43,7 +43,12 @@ protected override string GetTitle(string diagnosticId, string nodeText)
 
         protected override SyntaxNode AddSimplificationAnnotationTo(SyntaxNode expressionSyntax)
         {
-            var annotatedexpressionSyntax = expressionSyntax.WithAdditionalAnnotations(Simplifier.Annotation, Formatter.Annotation);
+            // Add the DoNotAllowVarAnnotation annotation.  All the code fixer
+            // does is pass the tagged node to the simplifier.  And we do *not*
+            // ever want the simplifier to produce 'var' in the 'Simplify type
+            // names' fixer.  only the 'Use var' fixer should produce 'var'.
+            var annotatedexpressionSyntax = expressionSyntax.WithAdditionalAnnotations(
+                Simplifier.Annotation, Formatter.Annotation, DoNotAllowVarAnnotation.Annotation);
```

Comment:
note: the fact that SimplifyTypeNames works by *acutally* simplifying during analysis, and then doing and just rerunning the full simplifier is pretty awful.  I intend to clean this up a lot in a later PRs.  For now, this is a nice and simple way to say: SimplifyTypeNames should not produce var, even if it thinks it can simplify the name of something.